### PR TITLE
Bump faraday and faraday_middleware gem

### PR DESCRIPTION
**Pull request by anaprimawaty**
### Description
Bump `faraday` and `faraday_middleware` from `~> 0` to `~> 1`. This is to resolve conflicts in gem dependencies, for example in `graphlient` ([gemspec link here](https://github.com/ashkan18/graphlient/blob/master/graphlient.gemspec#L17)).

I've done simple smoke testing using `bin/console` to query and create an account.

### Risks
**Medium**

### CODE REVIEW

**Comment by thaivz** [comment]
Looks good. Need `iron_bank` maintainers to comment.

**Comment by mickaelpham** [comment]
Thank you @anaprimawaty for opening this pull request. I will review the changes around `faraday` before merging. Please note that this will likely be a new major release (v5) as it might bring in breaking changes for the consumers of the gem (if anything, the fact that they would need to upgrade `faraday` too if they are using it for a different application).

File: lib/iron_bank/version.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
 module IronBank
-  VERSION     = "4.4.1"
+  VERSION     = "4.4.2"
```
Comment (mickaelpham) [inline lib/iron_bank/version.rb:4]: This gem follows [semantic versioning](https://semver.org/). As such, this change will likely be a major bump, not a patch bump.

**Comment by mickaelpham** [comment]
@anaprimawaty and @thaivz, thank you for your contribution. `IronBank v5.0.0` has been successfully built and released on [RubyGems.org](https://rubygems.org/gems/iron_bank).

**Comment by anaprimawaty** [comment]
Thanks for your help @mickaelpham !

**Comment by thaivz** [comment]
Thank you @mickaelpham !

**Pull request merged**